feat: support array type

kaori-seasons opened the pull request:
# Proposed changes

Issue Number: Related to [issue-341](https://github.com/apache/doris-spark-connector/issues/341)

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

JNSimba commented:
Thank you for your contribution.
1. Is the inferred data type inaccurate?
2. Currently, we are converting the Array to a String for reading. Have we encountered any problems with this method?

kaori-seasons commented:
> Thank you for your contribution.
> 
> 1. Is the inferred data type inaccurate?
> 2. Currently, we are converting the Array to a String for reading. Have we encountered any problems with this method?

Hello, thank you for your attention. In fact, I have already completed the conversion from array to string in the first commit of this PR and implemented relevant fallback strategies. However, after careful consideration, I believe that in scenarios with large amounts of data, type inference with arrow can reduce some performance overhead. What are your thoughts on this issue?

As shown in the PR, I've added a lot of integration tests to ensure that type inference is successful. Do you have any suggestions for this?
